git/config.py

```diff
@@ -443,7 +452,49 @@ def string_decode(v):
             raise e
 
     def _has_includes(self):
-        return self._merge_includes and self.has_section('include')
+        return self._merge_includes and len(self._included_paths())
+
+    def _included_paths(self):
+        """Return all paths that must be included to configuration.
+        """
+        paths = []
+
+        for section in self.sections():
+            if section == "include":
+                paths += self.items(section)
+
+            match = CONDITIONAL_INCLUDE_REGEXP.search(section)
+            if match is not None and self._repo is not None:
+                keyword = match.group(1)
+                value = match.group(2).strip()
+
+                if keyword in ["gitdir", "gitdir/i"]:
+                    value = osp.expanduser(value)
+
+                    if not any(value.startswith(s) for s in ["./", "/"]):
+                        value = "**/" + value
+                    if value.endswith("/"):
+                        value += "**"
+
+                    # Ensure that glob is always case insensitive if required.
+                    if keyword.endswith("/i"):
+                        value = re.sub(
+                            r"[a-zA-Z]",
+                            lambda m: "[{}{}]".format(
+                                m.group().lower(),
```

Comment:
Just as a remark: I really don't know how lower casing works in a unicode world, and if any of this is safe to do on paths. It's incredibly hard to get right, especially in Python.

Reply:
I think it is shouldn't be an issue as Python should be abstracting the complexity for us
https://docs.python.org/3.8/library/stdtypes.html#str.lower
https://docs.python.org/3.8/library/stdtypes.html#str.upper

Did you ever run into any issue with these two methods? Maybe in earlier Python 3 versions?

Reply:
Maybe at the time this code runs, it's already in the clear as it already is using a `str` which should have a known encoding.
The whole encoding story in GitPython is so thoroughly broken also due to Python making it so easy to get wrong that I simply don't trust any of it, which probably is a overly generic 😅.